src/base/bittorrent/session.cpp

```diff
@@ -2387,6 +2386,14 @@ void Session::setBannedIPs(const QStringList &newList)
                         , Log::WARNING);
         }
     }
+    // now we have to sort IPs and make them unique
+    filteredList.sort();
+    filteredList.removeDuplicates();
```

Comment:
Rewrote more code. Now it double checks for inequality of lists. Also added code for recreation of internal libtorrent ip filter from scratch to implement "unban" case.
But i found that manual ban list is enforced to ipfilterlist even when IPFiltering checkbox is disabled. I sink it is an error, isn't it?